File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveInt.cs
//{ "engine_register_objects":["Starlite__Raven__RavenAnimationDataCurveInt"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenAnimationDataCurveInt
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenAnimationDataCurveInt")]
        public sealed partial class RavenAnimationDataCurveInt : global::Starlite.Raven.RavenAnimationDataCurve<int> {
        }
        
    }
    
}
#pragma warning restore
#else
// RavenAnimationDataCurveInt
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenAnimationDataCurveInt, RavenAnimationDataCurveInt, RavenAnimationDataCurveInt, Starlite::Raven::RavenAnimationDataComponentBase, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveInt, m_Curves, m_Curves, kArray, kObject, Starlite::AnimationCurve::TypeId, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveInt, m_EaseType, m_EaseType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveInt, m_ValueType, m_ValueType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveInt, m_RepeatCount, m_RepeatCount, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveInt, m_Mirror, m_Mirror, kBool, kUnknown, -1, "")
SIMPLEMENT_CLASS_END(RavenAnimationDataCurveInt)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenAnimationDataCurveInt, Starlite::Raven, RavenAnimationDataCurveInt)
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveInt.cc
#ifdef STARLITE
#include "RavenAnimationDataCurveInt.h"
#include "RavenAnimationDataCurveInt.cs"

namespace Starlite {
    namespace Raven {
        RavenAnimationDataCurveInt::RavenAnimationDataCurveInt() {
        }

        int RavenAnimationDataCurveInt::EvaluateAtTime(double time, double duration) {
            return (int)m_Curves[0]->Evaluate((float)GetNormalizedTime(GetCurrentTime(time, duration), duration, m_EaseType));
        }

        void RavenAnimationDataCurveInt::SetStartingValues(const int& values) {
            m_Curves = Array<Ref<AnimationCurve>>(1);
            m_Curves[0] = Object::New<AnimationCurve>();
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(0, (float)values));
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(1, (float)values));
        }

        void RavenAnimationDataCurveInt::SyncStartingValues(const int& values) {
            auto& curve = m_Curves[0];
            auto& key = curve->GetKeyframes()[0];
            key.value = (float)values;
        }
    } // namespace Raven
} // namespace Starlite
#endif